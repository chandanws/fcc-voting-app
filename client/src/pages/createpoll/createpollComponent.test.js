import React from "react";
import CreatePollComponent from "./createpollComponent";
import { shallow } from "enzyme";

describe("<CreatePollComponent />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CreatePollComponent />);
    wrapper.setState({ options: ["option1", "option2", "option3"] });
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleOptionsChange", () => {
    it("should change options state based on option change", () => {
      wrapper.instance().handleOptionsChange(0)({
        target: { value: "newOption1" }
      });
      wrapper.update();
      expect(wrapper.state().options[0]).toBe("newOption1");
      expect(wrapper.state().options).toEqual([
        "newOption1",
        "option2",
        "option3"
      ]);
    });

    it("should not change state when index too high", () => {
      wrapper.instance().handleOptionsChange(7)({
        target: { value: "newOption1" }
      });
      wrapper.update();
      expect(wrapper.state().options).toEqual([
        "option1",
        "option2",
        "option3"
      ]);
    });
  });

  describe("handleTitleChange", () => {});

  describe("removeOption", () => {
    it("should remove the specified option by index", () => {
      wrapper.instance().removeOption(0);
      wrapper.update();
      expect(wrapper.state().options).toEqual(["option2", "option3"]);
    });
  });

  describe("addOption", () => {
    it("should add new option at the end of array", () => {
      wrapper.instance().addOption();
      wrapper.update();
      expect(wrapper.state().options.length).toBe(4);
      expect(wrapper.state().options[3]).toBe("");
    });
  });

  describe("handleSubmit", () => {});
});
