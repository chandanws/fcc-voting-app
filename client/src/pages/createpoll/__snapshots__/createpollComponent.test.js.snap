// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePollComponent /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreatePollComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Typography
          component="h1"
          text="Login"
/>,
        <Form
          id={1}
          onSubmit={[Function]}
>
          <Unknown
                    id={1}
                    label="title"
                    onChange={[Function]}
                    type="text"
                    value={undefined}
          />
          <div
                    className="form-wrapper--remove"
          >
                    <Unknown
                              id={0}
                              label="option"
                              onChange={[Function]}
                              type="text"
                              value="option1"
                    />
                    <button
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "alignSelf": "center",
                                                  "backgroundColor": "lightgrey",
                                                }
                              }
                    >
                              x
                    </button>
          </div>
          <div
                    className="form-wrapper--remove"
          >
                    <Unknown
                              id={1}
                              label="option"
                              onChange={[Function]}
                              type="text"
                              value="option2"
                    />
                    <button
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "alignSelf": "center",
                                                  "backgroundColor": "lightgrey",
                                                }
                              }
                    >
                              x
                    </button>
          </div>
          <div
                    className="form-wrapper--remove"
          >
                    <Unknown
                              id={2}
                              label="option"
                              onChange={[Function]}
                              type="text"
                              value="option3"
                    />
                    <button
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "alignSelf": "center",
                                                  "backgroundColor": "lightgrey",
                                                }
                              }
                    >
                              x
                    </button>
          </div>
          <button
                    onClick={[Function]}
                    type="button"
          >
                    Add option
          </button>
</Form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": "h1",
          "text": "Login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown
              id={1}
              label="title"
              onChange={[Function]}
              type="text"
              value={undefined}
/>,
            Array [
              <div
                className="form-wrapper--remove"
>
                <Unknown
                                id={0}
                                label="option"
                                onChange={[Function]}
                                type="text"
                                value="option1"
                />
                <button
                                onClick={[Function]}
                                style={
                                                Object {
                                                                "alignSelf": "center",
                                                                "backgroundColor": "lightgrey",
                                                              }
                                }
                >
                                x
                </button>
</div>,
              <div
                className="form-wrapper--remove"
>
                <Unknown
                                id={1}
                                label="option"
                                onChange={[Function]}
                                type="text"
                                value="option2"
                />
                <button
                                onClick={[Function]}
                                style={
                                                Object {
                                                                "alignSelf": "center",
                                                                "backgroundColor": "lightgrey",
                                                              }
                                }
                >
                                x
                </button>
</div>,
              <div
                className="form-wrapper--remove"
>
                <Unknown
                                id={2}
                                label="option"
                                onChange={[Function]}
                                type="text"
                                value="option3"
                />
                <button
                                onClick={[Function]}
                                style={
                                                Object {
                                                                "alignSelf": "center",
                                                                "backgroundColor": "lightgrey",
                                                              }
                                }
                >
                                x
                </button>
</div>,
            ],
            <button
              onClick={[Function]}
              type="button"
>
              Add option
</button>,
          ],
          "id": 1,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": 1,
              "label": "title",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown
                  id={0}
                  label="option"
                  onChange={[Function]}
                  type="text"
                  value="option1"
/>,
                <button
                  onClick={[Function]}
                  style={
                                    Object {
                                                      "alignSelf": "center",
                                                      "backgroundColor": "lightgrey",
                                                    }
                  }
>
                  x
</button>,
              ],
              "className": "form-wrapper--remove",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": 0,
                  "label": "option",
                  "onChange": [Function],
                  "type": "text",
                  "value": "option1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "x",
                  "onClick": [Function],
                  "style": Object {
                    "alignSelf": "center",
                    "backgroundColor": "lightgrey",
                  },
                },
                "ref": null,
                "rendered": "x",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown
                  id={1}
                  label="option"
                  onChange={[Function]}
                  type="text"
                  value="option2"
/>,
                <button
                  onClick={[Function]}
                  style={
                                    Object {
                                                      "alignSelf": "center",
                                                      "backgroundColor": "lightgrey",
                                                    }
                  }
>
                  x
</button>,
              ],
              "className": "form-wrapper--remove",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": 1,
                  "label": "option",
                  "onChange": [Function],
                  "type": "text",
                  "value": "option2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "x",
                  "onClick": [Function],
                  "style": Object {
                    "alignSelf": "center",
                    "backgroundColor": "lightgrey",
                  },
                },
                "ref": null,
                "rendered": "x",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown
                  id={2}
                  label="option"
                  onChange={[Function]}
                  type="text"
                  value="option3"
/>,
                <button
                  onClick={[Function]}
                  style={
                                    Object {
                                                      "alignSelf": "center",
                                                      "backgroundColor": "lightgrey",
                                                    }
                  }
>
                  x
</button>,
              ],
              "className": "form-wrapper--remove",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": 2,
                  "label": "option",
                  "onChange": [Function],
                  "type": "text",
                  "value": "option3",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "x",
                  "onClick": [Function],
                  "style": Object {
                    "alignSelf": "center",
                    "backgroundColor": "lightgrey",
                  },
                },
                "ref": null,
                "rendered": "x",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add option",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Add option",
            "type": "button",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Typography
            component="h1"
            text="Login"
/>,
          <Form
            id={1}
            onSubmit={[Function]}
>
            <Unknown
                        id={1}
                        label="title"
                        onChange={[Function]}
                        type="text"
                        value={undefined}
            />
            <div
                        className="form-wrapper--remove"
            >
                        <Unknown
                                    id={0}
                                    label="option"
                                    onChange={[Function]}
                                    type="text"
                                    value="option1"
                        />
                        <button
                                    onClick={[Function]}
                                    style={
                                                Object {
                                                            "alignSelf": "center",
                                                            "backgroundColor": "lightgrey",
                                                          }
                                    }
                        >
                                    x
                        </button>
            </div>
            <div
                        className="form-wrapper--remove"
            >
                        <Unknown
                                    id={1}
                                    label="option"
                                    onChange={[Function]}
                                    type="text"
                                    value="option2"
                        />
                        <button
                                    onClick={[Function]}
                                    style={
                                                Object {
                                                            "alignSelf": "center",
                                                            "backgroundColor": "lightgrey",
                                                          }
                                    }
                        >
                                    x
                        </button>
            </div>
            <div
                        className="form-wrapper--remove"
            >
                        <Unknown
                                    id={2}
                                    label="option"
                                    onChange={[Function]}
                                    type="text"
                                    value="option3"
                        />
                        <button
                                    onClick={[Function]}
                                    style={
                                                Object {
                                                            "alignSelf": "center",
                                                            "backgroundColor": "lightgrey",
                                                          }
                                    }
                        >
                                    x
                        </button>
            </div>
            <button
                        onClick={[Function]}
                        type="button"
            >
                        Add option
            </button>
</Form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": "h1",
            "text": "Login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown
                id={1}
                label="title"
                onChange={[Function]}
                type="text"
                value={undefined}
/>,
              Array [
                <div
                  className="form-wrapper--remove"
>
                  <Unknown
                                    id={0}
                                    label="option"
                                    onChange={[Function]}
                                    type="text"
                                    value="option1"
                  />
                  <button
                                    onClick={[Function]}
                                    style={
                                                      Object {
                                                                        "alignSelf": "center",
                                                                        "backgroundColor": "lightgrey",
                                                                      }
                                    }
                  >
                                    x
                  </button>
</div>,
                <div
                  className="form-wrapper--remove"
>
                  <Unknown
                                    id={1}
                                    label="option"
                                    onChange={[Function]}
                                    type="text"
                                    value="option2"
                  />
                  <button
                                    onClick={[Function]}
                                    style={
                                                      Object {
                                                                        "alignSelf": "center",
                                                                        "backgroundColor": "lightgrey",
                                                                      }
                                    }
                  >
                                    x
                  </button>
</div>,
                <div
                  className="form-wrapper--remove"
>
                  <Unknown
                                    id={2}
                                    label="option"
                                    onChange={[Function]}
                                    type="text"
                                    value="option3"
                  />
                  <button
                                    onClick={[Function]}
                                    style={
                                                      Object {
                                                                        "alignSelf": "center",
                                                                        "backgroundColor": "lightgrey",
                                                                      }
                                    }
                  >
                                    x
                  </button>
</div>,
              ],
              <button
                onClick={[Function]}
                type="button"
>
                Add option
</button>,
            ],
            "id": 1,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": 1,
                "label": "title",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown
                    id={0}
                    label="option"
                    onChange={[Function]}
                    type="text"
                    value="option1"
/>,
                  <button
                    onClick={[Function]}
                    style={
                                        Object {
                                                            "alignSelf": "center",
                                                            "backgroundColor": "lightgrey",
                                                          }
                    }
>
                    x
</button>,
                ],
                "className": "form-wrapper--remove",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": 0,
                    "label": "option",
                    "onChange": [Function],
                    "type": "text",
                    "value": "option1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "x",
                    "onClick": [Function],
                    "style": Object {
                      "alignSelf": "center",
                      "backgroundColor": "lightgrey",
                    },
                  },
                  "ref": null,
                  "rendered": "x",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown
                    id={1}
                    label="option"
                    onChange={[Function]}
                    type="text"
                    value="option2"
/>,
                  <button
                    onClick={[Function]}
                    style={
                                        Object {
                                                            "alignSelf": "center",
                                                            "backgroundColor": "lightgrey",
                                                          }
                    }
>
                    x
</button>,
                ],
                "className": "form-wrapper--remove",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": 1,
                    "label": "option",
                    "onChange": [Function],
                    "type": "text",
                    "value": "option2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "x",
                    "onClick": [Function],
                    "style": Object {
                      "alignSelf": "center",
                      "backgroundColor": "lightgrey",
                    },
                  },
                  "ref": null,
                  "rendered": "x",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown
                    id={2}
                    label="option"
                    onChange={[Function]}
                    type="text"
                    value="option3"
/>,
                  <button
                    onClick={[Function]}
                    style={
                                        Object {
                                                            "alignSelf": "center",
                                                            "backgroundColor": "lightgrey",
                                                          }
                    }
>
                    x
</button>,
                ],
                "className": "form-wrapper--remove",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": 2,
                    "label": "option",
                    "onChange": [Function],
                    "type": "text",
                    "value": "option3",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "x",
                    "onClick": [Function],
                    "style": Object {
                      "alignSelf": "center",
                      "backgroundColor": "lightgrey",
                    },
                  },
                  "ref": null,
                  "rendered": "x",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add option",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Add option",
              "type": "button",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
